htdocs/product/inventory/inventory.php

```diff
@@ -978,8 +978,10 @@ function barcodeserialforproduct(tabproduct,index,element,barcodeproductqty,sele
 	$sql = 'SELECT id.rowid, id.datec as date_creation, id.tms as date_modification, id.fk_inventory, id.fk_warehouse,';
 	$sql .= ' id.fk_product, id.batch, id.qty_stock, id.qty_view, id.qty_regulated, id.fk_movement, id.pmp_real, id.pmp_expected';
 	$sql .= ' FROM '.MAIN_DB_PREFIX.'inventorydet as id';
+	$sql .= ' INNER JOIN '.MAIN_DB_PREFIX.'product p on p.rowid = id.fk_product';
+
 	$sql .= ' WHERE id.fk_inventory = '.((int) $object->id);
-	$sql .= ' ORDER BY id.rowid';
+	$sql .= ' ORDER BY p.ref';
```

Comment:
You must keep id.rowid as a second sort parameter or the sorting will give non presicable result (several lins with same p.ref may be listed with different positions for same page)